### ISSUE TITLE
punct edeprels don't match basic deps

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/UD_English-EWT#453]
It looks like #449 updated the basic tree only, not the enhanced graph. @martinpopel is there a way to tell the script to do both?

**Comment by AngledLuffa** [comment]
Heh, I would think we can actually parse the previous change and apply all of the deltas as long as the head of the enhanced dependency was the same as the head of the basic dependency

**Comment by AngledLuffa** [comment]
```
diff_chunks = []
with open(diffs_file) as fin:
    next_chunk = []
    for line in fin:
        if line.startswith("@@"):
            if next_chunk:
                diff_chunks.append(next_chunk)
                next_chunk = []
        next_chunk.append(line)
    if next_chunk:
        diff_chunks.append(next_chunk)


with open(orig_file) as fin:
    orig_document = fin.readlines()

for chunk_idx, diff_chunk in enumerate(diff_chunks):
    diff_lines = diff_chunk[0].strip()[2:-2].strip()
    pieces = diff_lines.split()
    assert len(pieces) == 2
    start_line, start_len = map(int, pieces[0].split(","))
    end_line, end_len = map(int, pieces[1].split(","))
    assert start_len == end_len
    assert start_line == -end_line
    if start_line in (-60822, -105228):
        # a couple weird special cases
        continue

    idx = -start_line
    remove_edits = []
    add_edits = []
    for line in diff_chunk[1:]:
        if line.startswith("-"):
            remove_edits.append((idx, line))
        elif line.startswith("+"):
            add_edits.append((idx, line))
        else:
            idx += 1

    for edit_idx, (remove_edit, add_edit) in enumerate(zip(remove_edits, add_edits)):
        assert remove_edit[0] == add_edit[0]
        edit_line_idx = remove_edit[0] + edit_idx
        remove_pieces = remove_edit[1].split("\t")
        add_pieces = add_edit[1].split("\t")
        assert remove_pieces[7] in ('punct', 'root'), "Line %d is not punct or root!" % edit_line_idx
        assert add_pieces[7] in ('punct', 'root'), "Line %d is not punct or root!" % edit_line_idx
        assert remove_pieces[8] == "%s:%s" % (remove_pieces[6], remove_pieces[7]), "Enhanced different at %d" % edit_line_idx

        # remove the leading +
        add_pieces[0] = add_pieces[0][1:]
        # update to match the new dependency
        add_pieces[8] = "%s:%s" % (add_pieces[6], add_pieces[7])
        new_line = "\t".join(add_pieces)
        orig_document[edit_line_idx-1] = new_line


with open(orig_file, "w", encoding="utf-8") as fout:
    for line in orig_document:
        fout.write(line)
```

**Comment by AngledLuffa** [comment]
I just ran that script on the three files at the time of @martinpopel 's change.  The result is here:

https://github.com/UniversalDependencies/UD_English-EWT/pull/454

but there's one sentence where it wasn't obvious what to do, because the enhanced dependencies had a punctuation dependency on a fake node:

https://github.com/UniversalDependencies/UD_English-EWT/pull/454/commits/87438d7282fbf91af09e0d654d5d50a4e65e8e4b

If I can get some help on fixing that tree, I'll clean up the change by rebasing on dev and then converting everything to the component files.

**Comment by martinpopel** [comment]
I forgot to use the [copy_to_enhanced](https://github.com/udapi/udapi-python/blob/master/udapi/block/ud/fixpunct.py#L56) parameter. I will send a PR soon.

**Comment by AngledLuffa** [comment]
@martinpopel I've already done that, I believe, with the exception of the one tree I highlighted in https://github.com/UniversalDependencies/UD_English-EWT/commit/87438d7282fbf91af09e0d654d5d50a4e65e8e4b

I do figure it would save some time to start from there, but it's also no big deal to me if you go back and redo it.

**Comment by nschneid** [comment]
Fixed in #456
